File: src/main/java/com/matchFit/post/entity/Post.java
package com.matchFit.post.entity;

import com.matchFit.common.entity.BaseEntity;
import com.matchFit.user.entity.Gender;
import com.matchFit.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Post extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, length = 300)
    private String description;

    @Column
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Gender gender;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Sports sports;

    @Column(nullable = false)
    private int cost;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private Status status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Town town;

    @Column(nullable = false)
    private int maxPeople;

    @Column(nullable = false)
    private LocalDateTime date;

    @Column(nullable = false)
    private String location;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: src/main/kotlin/com/matchFit/post/entity/Post.kt
package com.matchFit.post.entity

import com.matchFit.common.entity.BaseEntity
import com.matchFit.user.entity.Gender
import com.matchFit.user.entity.User
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime

@Entity
class Post : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(nullable = false)
    lateinit var title: String

    @Column(nullable = false, length = 300)
    lateinit var description: String

    @Column(nullable = true)
    var imageUrl: String? = null

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    lateinit var gender: Gender

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    lateinit var sports: Sports

    @Column(nullable = false)
    var cost: Int = 0

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    lateinit var status: Status

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    lateinit var town: Town

    @Column(nullable = false)
    var maxPeople: Int = 0

    @Column(nullable = false)
    lateinit var date: LocalDateTime

    @Column(nullable = false)
    lateinit var location: String

    @ManyToOne
    @JoinColumn(name = "user_id")
    lateinit var user: User
}
